code/modules/unit_tests/organs.dm

```diff
@@ -0,0 +1,174 @@
+ * Checks that all implantable DNA Infuser organs are set up correctly and without error.
+ * Ensures that the "organ_set_bonus" status effects activate and deactivate.
+ *
+ * "Life's about having a good time and having candy, not putting roaches in your hair!"
+ * - Dr. Steve Brule
+ */
+/// Ensures the organ inserted properly by validating specific variables.
+#define TEST_ORGAN_INSERTED_OK(lab_rat, test_organ) ((test_organ.owner == lab_rat) && (test_organ in lab_rat.internal_organs) && test_organ.slot ? (lab_rat.getorganslot(test_organ.slot) == test_organ) : TRUE)
+/// Ensures the organ removed properly by validating specific variables.
+#define TEST_ORGAN_REMOVED_OK(lab_rat, test_organ) ((test_organ.owner == null) && !(test_organ in lab_rat.internal_organs) && test_organ.slot ? (lab_rat.getorganslot(test_organ.slot) != test_organ) : TRUE)
+/// List of organ typepaths which are not test-able, such as certain class prototypes.
+#define TEST_ORGAN_BLACKLIST list(\
+	/obj/item/organ/internal,\
+	/obj/item/organ/external,\
+	/obj/item/organ/external/wings,\
+	/obj/item/organ/internal/cyberimp,\
+	/obj/item/organ/internal/cyberimp/brain,\
+	/obj/item/organ/internal/cyberimp/mouth,\
+	/obj/item/organ/internal/cyberimp/arm,\
+	/obj/item/organ/internal/cyberimp/chest,\
+	/obj/item/organ/internal/cyberimp/eyes,\
+	/obj/item/organ/internal/alien,\
+)
+
+//-- Sanity: ALL ORGANS --//
+/// Check organ insertion and removal, for all organ subtypes usable in-game.
+/datum/unit_test/organ_sanity/Run()
+	for(var/obj/item/organ/organ_type as anything in subtypesof(/obj/item/organ))
+		// Skip prototypes.
+		if(organ_type in TEST_ORGAN_BLACKLIST)
+			continue
+
+		// Appropriate mob (Human) which will receive organ.
+		var/mob/living/carbon/human/lab_rat = allocate(/mob/living/carbon/human/consistent)
+		var/obj/item/organ/test_organ = new organ_type()
+
+		// Inappropriate mob (Dog) which will hopefully reject organ.
+		var/mob/living/basic/pet/dog/lab_dog = allocate(/mob/living/basic/pet/dog/corgi)
+		var/obj/item/organ/reject_organ = new organ_type()
+
+		// Insert organ and store status code in var.
+		var/inserted_ok = test_organ.Insert(lab_rat, special = TRUE, drop_if_replaced = FALSE)
+		// Ensure organ was rejected.
+		var/rejected_ok = !reject_organ.Insert(lab_dog, special = TRUE, drop_if_replaced = FALSE)
+
+		// For the Human, expects status code 1
+		if(!inserted_ok)
+			TEST_FAIL("The organ \"[test_organ.type]\" was not inserted in the mob when expected, Insert() returned FALSE when TRUE was expected.")
+			continue
+
+		// For the Dog, expects status code 0 or falsy value.
+		if(!rejected_ok)
+			TEST_FAIL("The \"[test_organ.type]\" was inserted into a basic mob (Corgi) when it wasn't expected.")
+
+		// Inserting Nightmare brain causes the Human's species to change.
+		// Species change swaps out all the organs, making test_organ un-usable by this point.
+		if(test_organ.type == /obj/item/organ/internal/brain/shadow/nightmare)
```

Comment:
We should have a separate typecache for this instead of hardcoding it directly

Reply:
How do I use a type cache for this line? This doesn't seem like the usual use case for type cache lists